lib/SimpleHttpClient/Communicator.cpp

```diff
@@ -440,10 +440,21 @@ void Communicator::handleResult(CURL* handle, CURLcode rc) {
     return;
   }
 
+  // unclear if this would be safe on another thread.  Leaving here.
   if (rip->_options._curlRcFn) {
     (*rip->_options._curlRcFn)(rc);
   }
 
+  auto inProgress = _handlesInProgress.find(rip->_ticketId);
```

Comment:
Looks unsafe, as this accesses `_handlesInProgress` without the mutex (which is acquired a few lines below).

Reply:
Agreed.  Starting new branch with correction.  This one already merged.